### ISSUE TITLE
[BUG] Get-PnPTeamsTeam: The request uses a filter property that is not indexed

### ISSUE THREAD

**Issue by YuriySamorodov** [pnp/powershell#2721]
### Reporting an Issue or Missing Feature
I am reporting a bug

### Expected behavior 
`Get-PnPTeamsTeam -Filtet ""createdDateTime ge 2023-01-14T00:00:00Z"` should return groups created since January 14th, 2023 as per the articles below:

[Advanced query capabilities on Azure AD directory objects](https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http#group-properties)

Other users report the same issue:
[https://learn.microsoft.com/en-us/answers/questions/499031/azure-graph-api-for-users-doesnt-support-filter-by](https://learn.microsoft.com/en-us/answers/questions/499031/azure-graph-api-for-users-doesnt-support-filter-by)
### Actual behavior 
When I run `Get-PnPTeamsTeam -Filtet ""createdDateTime ge 2023-01-14T00:00:00Z"` command I get the following error:

> Get-PnPTeamsTeam: Bad Request (400): The request uses a filter property that is not indexed

![image](https://user-images.githubusercontent.com/5260172/212549720-33045b50-e4d8-49e1-8406-4725fbe5040e.png)

### Steps to reproduce behavior
```
Connect-PnPOnline
Get-PnPTeamsTeam -Filtet ""createdDateTime ge 2023-01-14T00:00:00Z"
```
### What is the version of the Cmdlet module you are running?
```
Name    : PnP.PowerShell
Version : 1.12.64
```

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by KoenZomers** [comment]
Looks like this is as designed? Its literally used as an example where it would/show this response:
https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http#group-properties:~:text=If%20a%20property%20has%20not%20been%20indexed%20to%20support%20a%20query%20parameter%2C%20even%20if%20the%20advanced%20query%20parameters%20are%20specified%2C%20the%20request%20returns%20an%20error

Its also not mentioned as supported on a Group:
https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http#group-properties

Correct me if I'm wrong.

**Comment by smapelli** [comment]
Documentation is not consistent as filtering should be allowed based on https://learn.microsoft.com/en-us/graph/api/resources/group?view=graph-rest-1.0#properties

![image](https://user-images.githubusercontent.com/40613147/214802771-6807875f-d1a4-44a6-bb82-a2b4bd0bf762.png)

**Comment by KoenZomers** [comment]
Agree, but that's not something the PnP team can do something about :) Please file a support ticket with Microsoft Support if you would like filtering on the createdDateTime to be added or fixed on the Microsoft Graph side. From the PnP perspective we're just passing onwards the query and that part seems to work well.

**Comment by smapelli** [comment]
As an alternative to createdDateTime, filtering could be done on renewedDateTime (and it works as I've just now tested it) which, if M365 groups expiration is not enabled, should always carry the same timestamp that is in createdDateTime.

**Comment by gautamdsheth** [comment]
Ok closing this issue now as the issue lies with Graph API. Once the API starts supporting filter on the `createdDateTime` it will start working. The fix needs to be done at Microsoft's end and we can't do anything about it here.

Having said that, would also suggest using `Get-PnPMicrosoft365ExpiringGroup` if you want to search for groups that are nearing expiration.
